sympy/polys/matrices/ddm.py

```diff
@@ -1000,6 +1077,64 @@ def qr(self):
 
         return Q, R
 
+    def fraction_free_qrd(self):
+        """
+        Compute a fraction-free QR decomposition for DDM.
```

Comment:
Where does this algorithm come from?

It doesn't use the PLDU decomposition. I thought that the idea here was to use the fraction-free PLDU to compute the fraction-free QRD.